libs/viewer/src/SimpleViewer.cpp

```diff
@@ -1625,6 +1634,22 @@ void SimpleViewer::updateUserInterface() {
         lm.setShadowCaster(ci, light.enableShadows);
     });
 
+    if (ImGui::CollapsingHeader("Triplanar setup")) {
+        static float yaw = 0.0f;
+        static float pitch = 0.0f;
+        static float roll = 0.0f;
+        ImGui::SliderAngle("Yaw (Y)", &yaw);
+        ImGui::SliderAngle("Pitch (X)", &pitch);
+        ImGui::SliderAngle("Roll (Z)", &roll);
+
+        // We are setting the inverse of the triplanar transform here so that the shader does not have
+        // to deal with the back-and-forth transforms between spaces (as triplanar computations will be
+        // carried out in the local coordinate system of the triplanar box). 
+        triplanarTransform = filament::math::mat3f::eulerYXZ(-yaw, -pitch, -roll);
+        triplanarNormalTransform = filament::math::mat3f::getTransformForNormals(triplanarTransform);
```

Comment:
Why is `triplanarNormalTransform` needed? `triplanarTransform` doesn't contain non-uniform scale, it is just a plain rotation matrix, safely invertible and usable for normal vectors too. If this is the optimization you mentioned in the PR description I'd say go for it!

Moreover, I'd call it `triplanarRotation` to further emphasize that it's just a rotation.

Reply:
Yes, that's the optimization I'm aiming for (and the one Refractive already has to live with as we've ran out of uniform buffer space). We'll see how orthonormal the frames are and if those are a go, this will be simplified.